std/string.c3

```diff
@@ -1,10 +1,29 @@
 @require_once "arithmetic.c3"
 @require_once "assert.c3"
+@require_once "iterators.c3"
 @require_once "util.c3"
 @require_once "wrappers.c3"
 
 typedef StringView : Span<u8>
 
+@operator == : (lhs : StringView, rhs : StringView) -> bool = {
```

Comment:
Wasn't `StringView` a `Span<u8>`? Can't we then generalize this for `Span<T>`?

Reply:
Good point. That's seems useful. We don't have the SFINAE constructs to limit it to spans with an elementwise `==` but it'll be fine for now.

Reply:
Maybe this should count as a "substitution failure" for the purposes of SFINAE